dynamodb.py

```diff
@@ -0,0 +1,219 @@
+
+    def _find_usage_dynamodb(self):
+        ''' calculates current usage for all DynamoDB limits '''
+
+        logger.debug("Getting usage for DynamoDB tables")
+        
+        table_count = 0
+        region_read_capacity = 0
+        region_write_capacity = 0
+
+        for table in self.connect().list_tables()['TableNames']:
+
+            table_count += 1
+            gsi_write = 0
+            gsi_read = 0
+            table_Arn = self.connect().describe_table(TableName=table)['Table']['TableArn']
+            UsagePerTable = self.connect().describe_table(TableName=table)['Table']
+
+            logger.debug("Getting usage for DynamoDB tables")
+
+            for gsi in UsagePerTable['GlobalSecondaryIndexes']:
+                gsi_read += gsi['ProvisionedThroughput']['ReadCapacityUnits']
+                gsi_write += gsi['ProvisionedThroughput']['WriteCapacityUnits']
+
+            table_write_capacity = UsagePerTable['ProvisionedThroughput']['WriteCapacityUnits'] + gsi_write
+            table_read_capacity = UsagePerTable['ProvisionedThroughput']['ReadCapacityUnits'] + gsi_read
+            region_read_capacity += table_read_capacity
+            region_write_capacity += table_write_capacity
+
+            self.limits['Global Secondary Indexes (table)']._add_current_usage(
+                len(UsagePerTable['GlobalSecondaryIndexes'] if 'GlobalSecondaryIndexes' in UsagePerTable else 0),
+                resource_id=table_Arn, aws_type='AWS::DynamoDB::Table'
+            )
+            self.limits['Local Secondary Indexes (table)']._add_current_usage(
+                len(UsagePerTable['LocalSecondaryIndexes'] if 'LocalSecondaryIndexes' in UsagePerTable else 0),
+                resource_id=table_Arn, aws_type='AWS::DynamoDB::Table'
+            )
+            self.limits['Write Capacity (table)']._add_current_usage(
+                table_write_capacity,
+                resource_id=table_Arn
+            )
+            self.limits['Read Capacity (table)']._add_current_usage(
+                table_read_capacity,
+                resource_id=table_Arn
+            )
+
+        self.limits['Table Count (region)']._add_current_usage(
+            table_count,
+            aws_type='AWS::DynamoDB::Table'
+        )
+        self.limits['Write Capacity (region)']._add_current_usage(
+            region_write_capacity,
+            aws_type='AWS::DynamoDB::Table'
+        )
+        self.limits['Read Capacity (region)']._add_current_usage(
+            region_read_capacity,
+            aws_type='AWS::DynamoDB::Table'
+        )
+
+    def get_limits(self):
+        """
+        Return all known limits for this service, as a dict of their names
+        to :py:class:`~.AwsLimit` objects.
+        :returns: dict of limit names to :py:class:`~.AwsLimit` objects
+        :rtype: dict
+        """
+        if self.limits != {}:
+            return self.limits
+        limits = {}
+        limits.update(self._get_limits_dynamodb())
```

Comment:
If all this method does is call ``_get_limits_dynamodb()``, they should be one method not two.